kiosk/src/Transforms/gameOver.ts

```diff
@@ -27,7 +27,7 @@ export function gameOver(skipHighScore?: boolean): void {
             selectedGame?.highScoreMode !== "None" &&
             state.mostRecentScores?.length
         ) {
-            exitToEnterHighScore();
+            exitToEnterHighScore(selectedGame?.highScoreMode!);
```

Comment:
hmm, if `selectedGame` is undefined here, this would cause an issue. I'm not sure what would cause it to end up null, but should we just have a check for that case above and handle it there, rather than having this optional chaining followed by a non-null assertion, which seems sort of contradictory.

Reply:
Yeah, that's a good idea. This was something I was thinking about yesterday as well, but I figured since we weren't doing the explicit check before that it would be fine still, but it's definitely better to be safer here. And it's very true that it will be cleaner to not have to do the optional chaining and assertion business here.

Reply:
Before, it looks like we didn't rely on selectedGame to actually provide any required information, but with this change we need highScoreMode to be defined. I'd recommend just putting a `selectedGame !== undefined` before `selectedGame?.highScoreMode !== "None"` and removing the `?` from the selectedGame references following that. That way, it'll just default to exitGame when selectedGame is undefined.